### PR TITLE
REST API: Add autoupdates log endpoint

### PR DESCRIPTION
**Pull request by lezama**

### CODE REVIEW

File: class.jetpack-sync.php [hunk 1]
```diff
@@ -1101,4 +1103,19 @@ static function get_constant( $constant ) {
...
+			return;
+		}
+		$log = Jetpack_Options::get_option( 'updates_log', array() );
+		// Append our event to the log
+		$log[] = array(
+			'time'    => time(),
+			'user_id' => get_current_user_id(),
+			'results'    => $results,
+		);
+		Jetpack_Options::update_option( 'updates_log', $log );
```
Comment (roccotripaldi) [inline class.jetpack-sync.php]: We've already been creating a log here:
https://github.com/Automattic/jetpack/blob/master/class.jetpack-autoupdate.php#L111

Perhaps the endpoint could be used to grab that? Or at the very least we should stop the old log in favor of the new, so we are not saving things twice.

Comment (lezama) [inline class.jetpack-sync.php]: Decided via hangout, we are going to merge both approaches :+1:

File: class.jetpack-autoupdate.php [hunk 2]
```diff
@@ -87,29 +88,29 @@ function expect( $item, $type='plugin' ) {
...
-	/**
-	 * On shutdown, let's check to see if we've preformed an automatic update.
-	 * If so, let's compare the expected results to the actual results, and log our findings.
-	 *
-	 * Results are logged locally via Jetpack::log(), and globally via Jetpack::do_stats()
-	 */
-	function log_results() {
 
-		if ( $this->is_updating ) {
+		if ( empty( $results ) && empty( $this->expected ) ) {
```
Comment (lezama) [inline class.jetpack-autoupdate.php]: `$this->expected` will never be empty!

File: class.jetpack.php [hunk 3]
```diff
@@ -2981,7 +2981,8 @@ public static function try_registration() {
 	 * [Everyone Loves a Log!](https://www.youtube.com/watch?v=2C7mNr5WMjA)
 	 */
 	public static function log( $code, $data = null ) {
-		$log = Jetpack_Options::get_option( 'log', array() );
+		// only grab the latest 200 entries
```
Comment (enejb) [inline class.jetpack.php]: only keep the last 200.

**Comment by enejb** [comment]
Should we allow devs to specify the limit?

**Comment by enejb** [comment]
Also I am getting the following php notices 
```[19-Nov-2015 19:18:42 UTC] PHP Deprecated:  Non-static method Jetpack::stat() should not be called statically, assuming $this from incompatible context in /home/enej/public_html/aaa/wp-content/plugins/jetpack/class.jetpack-autoupdate.php on line 140
[19-Nov-2015 19:18:42 UTC] PHP Deprecated:  Non-static method Jetpack::do_stats() should not be called statically, assuming $this from incompatible context in /home/enej/public_html/aaa/wp-content/plugins/jetpack/class.jetpack-autoupdate.php on line 159
[19-Nov-2015 19:21:33 UTC] PHP Deprecated:  Non-static method Jetpack::stat() should not be called statically, assuming $this from incompatible context in /home/enej/public_html/aaa/wp-content/plugins/jetpack/class.jetpack-autoupdate.php on line 140
[19-Nov-2015 19:21:33 UTC] PHP Deprecated:  Non-static method Jetpack::do_stats() should not be called statically, assuming $this from incompatible context in /home/enej/public_html/aaa/wp-content/plugins/jetpack/class.jetpack-autoupdate.php on line 159
[19-Nov-2015 19:46:36 UTC] PHP Deprecated:  Non-static method Jetpack::stat() should not be called statically, assuming $this from incompatible context in /home/enej/public_html/aaa/wp-content/plugins/jetpack/class.jetpack-autoupdate.php on line 140
[19-Nov-2015 19:46:36 UTC] PHP Deprecated:  Non-static method Jetpack::do_stats() should not be called statically, assuming $this from incompatible context in /home/enej/public_html/aaa/wp-content/plugins/jetpack/class.jetpack-autoupdate.php on line 159```

**Comment by lezama** [comment]
> Should we allow devs to specify the limit that is higher then the 200? Maybe we just need to update the documentation on the method.

We could add a filter

**Pull request merged**